lib/Core/BuildEngine.cpp

```diff
@@ -816,9 +816,9 @@ class BuildEngineImpl {
           std::string error;
           bool result = db->setRuleResult(ruleInfo->rule, ruleInfo->result, &error);
           if (!result) {
-            // FIXME: Error handling.
-            std::cerr << error << std::endl;
-            exit(1);
+            delegate.error(error);
+            completeRemainingTasks();
+            return false;
```

Comment:
I was thinking about that, but probably not worth it right now, because the error condition itself isn't very common.